module/documents/actor/actor.mjs

```diff
@@ -3254,28 +3278,42 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
   /**
    * Flash ring & display changes to health as scrolling combat text.
-   * @param {object} options  Any data provided by the update.
+   * @param {object} changes          Object of changes to hit points.
+   * @param {number} changes.hp       Changes to `hp.value`.
+   * @param {number} changes.temp     The change to `hp.temp`.
+   * @param {number} changes.total    The total change to hit points.
    * @protected
    */
-  _displayTokenEffect(options) {
-    const dhp = Number(options.dhp);
+  _displayTokenEffect(changes) {
+    let key;
+    let change;
+    if ( !parseInt(changes.hp) && parseInt(changes.temp) ) key = "temp";
+    else {
+      key = "hp";
+      change = changes.total;
+    }
+    if ( !key ) return;
+    change ??= changes[key];
```

Comment:
If the change is only to temp hp (up or down), then only that should be displayed. If there is any change at all to hp.value, then show healing or damage.